### PR TITLE
Add attribute for custom events | add trait

### LINKED ISSUE #926: [Nox.Lib] Add To Custom IE the EventTypeAttribute
- relates to #799 

## Custom IE, Trait and Event name

1. Add a Trait (string type), required to the integration events in Nox.Solution
2. Update the Custom Integration Event Template and add the integration event attribute, example

  ```
     [IntegrationEvent("PopulationHigherThan","Country")]
     public record PopulationHigherThan(...): IIntegrationEvent;    
```

Nox.Solution example:

```
application:
  integrationEvents:
        - name: PopulationHigherThan
          type: object
          trait: Country
          objectTypeOptions:
            attributes:
              - name: CountryId
                type: countryCode2

              - name: CountryName
                type: text

              - name: Population
                type: number
```

### PR DESCRIPTION
**Pull request by ILoveMyTortoise**
closes #926

### CODE REVIEW

File: src/Nox.Solution/Validation/Application/Events/IntegrationEventValidator.cs [hunk 1]
```diff
@@ -15,6 +14,11 @@ public IntegrationEventValidator()
             .NotEmpty()
             .WithMessage(m => string.Format(ValidationResources.ApplicationEventTypeEmpty, m.Name));
 
+        // TODO: should be added or attribute is enough?
```
Comment (rochar) [inline src/Nox.Solution/Validation/Application/Events/IntegrationEventValidator.cs]: The required attribute is enough, you can remove this comment code, thks

**Review by rochar — DISMISSED** [review]
looks good, thks

remove the comment code and let me know it

File: src/Nox.Generator/Application/IntegrationEvents/CustomIntegrationEvent.template.cs [hunk 2]
```diff
@@ -12,13 +12,24 @@
 {{ attributes = integrationEvent.ObjectTypeOptions.Attributes -}}
 {{- end -}}
 // Generated
+{{func pascalCaseToCamelCase(pascal)
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Generator/Application/IntegrationEvents/CustomIntegrationEvent.template.cs:15]: To not have the same code in multiple places, we can maybe in the future try to export the functions in a separate file and import them if possible in the template we need to use.

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]
Approved, with a small comment which we can check in the future to improve it a bit.

File: src/Nox.Generator/Application/IntegrationEvents/CustomIntegrationEvent.template.cs (hunk 2 above)
Comment (ILoveMyTortoise) [reply src/Nox.Generator/Application/IntegrationEvents/CustomIntegrationEvent.template.cs:15]: yes, will look into this in the future

**Review by rochar — APPROVED** [review]

**Pull request merged**